### PR TITLE
improve command_reference.md (among other improvements)

### PR DESCRIPTION
**Pull request by timfeirg**
* 为 cache.md 补充内容
* 丰富 command_reference.md 内容

### CODE REVIEW

File: docs/zh_cn/reference/command_reference.md [hunk 1]
```diff
@@ -881,7 +879,7 @@ $ ACCESS_KEY=myAccessKey SECRET_KEY=mySecretKey juicefs objbench --storage s3  h
 
 #### 描述
 
-收集泄漏的对象。
+收集泄漏的对象。我们不认为 JuiceFS 会产生对象泄露（出现泄露说明有 Bug），如果你观察到对象存储用量异常，怀疑出现了对象泄露，可以使用该命令进行检查和处理。
```
Comment (davies) [inline docs/zh_cn/reference/command_reference.md:882]: 尽量用客户描述：JuiceFS 通常不会泄漏对象，...

File: docs/en/reference/command_reference.md [hunk 2]
```diff
@@ -878,7 +878,7 @@ $ ACCESS_KEY=myAccessKey SECRET_KEY=mySecretKey juicefs objbench --storage s3  h
 
 #### Description
 
-Collect and delete leaked objects. We don't expect JuiceFS to produce leak in object storage (only happens when there's bug in JuiceFS), use this command if you notice any abnormal usage in object storage.
+Collect and delete leaked objects. JuiceFS doesn't easily produce leak in object storage (only happens when there's bug in JuiceFS), use this command if you notice any abnormal usage in object storage.
```
Comment (davies) [inline docs/en/reference/command_reference.md:881]: usually

**Comment by timfeirg** [comment]
plan to further polish trash.md

**Comment by SandyXSD** [comment]
IMO this is a pretty good reference, we can learn from it:
https://docs.docker.com/engine/reference/commandline/run/

File: docs/zh_cn/reference/command_reference.md [hunk 3]
```diff
@@ -877,11 +875,11 @@ juicefs objbench [command options] BUCKET
 $ ACCESS_KEY=myAccessKey SECRET_KEY=mySecretKey juicefs objbench --storage s3  https://mybucket.s3.us-east-2.amazonaws.com -p 6
 ```
 
-### juicefs gc
+### juicefs gc {#gc}
 
 #### 描述
 
-收集泄漏的对象。
+收集泄漏的对象。JuiceFS 通常不会产生对象泄露（若出现泄露，说明存在 Bug），如果你观察到对象存储用量异常，怀疑出现了对象泄露，可以使用该命令进行检查和处理。
```
Comment (xiaogaozi) [inline docs/zh_cn/reference/command_reference.md:882]: 不完全只是用来收集泄漏的对象，还可以合并覆盖写产生的对象碎片，详见[这个文档](https://juicefs.com/docs/zh/community/administration/status_check_and_maintenance#gc)。

File: docs/zh_cn/reference/command_reference.md [hunk 4]
```diff
@@ -155,16 +151,16 @@ juicefs format [command options] META-URL NAME
 #### 选项
 
 `--block-size value`<br />
-块大小；单位为 KiB (默认：4096)
+块大小；单位为 KiB (默认：4096)。4M 是一个较好的默认值，不少对象存储（比如 S3）都将 4M 设为内部的块大小，因此将 JuiceFS block size 设为相同大小，往往也能获得更好的性能
```
[the remark this answers is not in the document]
Comment (timfeirg) [reply docs/zh_cn/reference/command_reference.md:154]: 内容均来自 liyong 当时给我随口分享的, 我确实也没找到合适的 ref, 我随手搜了下 "s3 4M block", 能查到一些类似的说法:

https://mariadb.com/kb/en/s3-storage-engine-internals/#big-reads

**Comment by timfeirg** [comment]
RFC @xiaogaozi

**Comment by davies** [comment]
LGTM overall.

**Review by davies — APPROVED** [review]

**Pull request merged**